keylime-push-model-agent/Cargo.toml

```diff
@@ -0,0 +1,56 @@
+[package]
+name = "keylime_push_model_agent"
+description = "Rust agent for Keylime (Push Model)"
+authors.workspace = true
+edition.workspace = true
+license.workspace = true
+repository.workspace = true
+version.workspace = true
+
+[dependencies]
+actix-web.workspace = true
+assert_cmd.workspace = true
+base64.workspace = true
+cfg-if.workspace = true
+clap.workspace = true
+config.workspace = true
+futures.workspace = true
+glob.workspace = true
+hex.workspace = true
+keylime.workspace = true
+libc.workspace = true
+log.workspace = true
+openssl.workspace = true
+predicates.workspace = true
+pretty_env_logger.workspace = true
+reqwest.workspace = true
+serde.workspace = true
+serde_derive.workspace = true
+serde_json.workspace = true
+static_assertions.workspace = true
+tempfile.workspace = true
+tokio.workspace = true
+tss-esapi.workspace = true
+thiserror.workspace = true
+uuid.workspace = true
+zeroize = "1.8.1"
+getrandom = "0.3.2"
```

Comment:
Could you please try to cleanup the dependencies here and remove things that you will not use?
Also please be very defensive and add new dependencies only when strictly needed. If you really need a dependency, please check its availability in Fedora first.

Reply:
Sorry, I forgot to groom this. I kept those strictly necessary.

By checking in Fedora ... do you mean the component is available in Fedora? Are dependencies not handled through Cargo? Why are tests passing in the different Fedora versions if packages are not available?

Reply:
I checked in Fedora 41 and both `assert_cmd` and `predicates` exist:
```bash
$ dnf search rust assert_cmd 2>/dev/null 1>/dev/null && echo exists
exists
$ dnf search rust predicates 2>/dev/null 1>/dev/null && echo exists
exists
```

Reply:
Yes, both exists:
https://src.fedoraproject.org/rpms/rust-predicates
https://src.fedoraproject.org/rpms/rust-assert_cmd

If possible, I would like to add them, as both packages enable testing binaries easily:
https://github.com/keylime/rust-keylime/pull/964/files#diff-629dea229153170f9db2325748def31f811e87e9cc49c68307740a97ca0d57e3R18